lib/matplotlib/axes/_base.py

```diff
@@ -1302,9 +1302,14 @@ def __clear(self):
         self._get_patches_for_fill = _process_plot_var_args('fill')
 
         self._gridOn = mpl.rcParams['axes.grid']
+        # Swap children to minimize time we spend in an invalid state
         old_children, self._children = self._children, []
         for chld in old_children:
-            chld.axes = chld._parent_figure = None
+            chld._parent_figure = None
+            chld.axes = None
+            chld._remove_method = None
```

Comment:
Thinking of potential future threading issues, you should probably set `chld._remove_method` first.
```suggestion
            chld._remove_method = None
            chld._parent_figure = None
            chld.axes = None
```